ext/src/Smallint.c

```diff
@@ -242,10 +242,6 @@ PHP_METHOD(Smallint, div)
     }
 
     result->data.smallint.value = self->data.smallint.value / smallint->data.smallint.value;
-    if (result->data.smallint.value * smallint->data.smallint.value != self->data.smallint.value) {
```

Comment:
My thoughts on why this was there was because the documentation indicates the arithmetic operation methods can take a [Numeric](https://github.com/datastax/php-driver/blob/master/ext/doc/Cassandra/Tinyint.php#L71)  value; however it really only takes the same type as the variable instance. Because of that, there is no possible value that would ever make that true except for when `result->data.smallint.value = 0` (e.g. 0/1) which is actually a bug.

I think we might want to create another ticket that allows the use of `int`|`double`|`string`|`instance_type` because it is a little cumbersome to force the use of just the instance type when the constructors for each type allow for all the other types and handle the conversion.